azavea/cac-tripplanner
# Feature/kak/app install banner#1069

## Overview

Modify app manifest to note Play store application. This will display a banner prompt on Chrome/Android to install the Play store app (instead of the progressive web app to homescreen) if neither are already installed. Also adds a 512px logo, as required, and corrects content type for app manifest.

[Native app install banners documentation](https://developers.google.com/web/fundamentals/app-install-banners/native)

### Demo

#### Banner prompt at bottom of screen:

![screenshot_20180813-123331](https://user-images.githubusercontent.com/960264/44046148-9af6d8b4-9ef8-11e8-8dd8-7a6097177e0d.png)

#### On banner tap:
![screenshot_20180813-123821](https://user-images.githubusercontent.com/960264/44046205-bb85c0d6-9ef8-11e8-90b2-2ae178eaa37d.png)

##### On tapping 'install':
![screenshot_20180813-123830](https://user-images.githubusercontent.com/960264/44046214-c063d0d4-9ef8-11e8-9764-070437b2198c.png)

#### After install:
![screenshot_20180813-123855](https://user-images.githubusercontent.com/960264/44046219-c5029e0e-9ef8-11e8-980f-5caa16e10e54.png)


## Notes

Regarding the Content Security Policy requirement noted in the documentation (link above), I believe the current setting of `default-src https` should allow all `img-src` sources, so we shouldn't need to modify our nginx configuration to support this in production.

As a side note, the [app manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest) property for related applications appears to be platform-agnostic, and so should also work if we add an iOS app later.

## Testing Instructions

 * `vagrant ssh app`
 * `cd /opt/app/src/ && npm run gulp-development`
 * Host local site over HTTPS via ngrok or similar (`ngrok http 8024`)
 * From an Android device: uninstall GoPhillyGo native app and/or web app, if installed
 * Visit the HTTPS URL on Chrome/Android
 * After interacting with the site for 30 seconds, install banner should appear at bottom of page
 * Install process should work as expected

Closes #1069
